plugin.py

```diff
@@ -91,25 +91,30 @@ def weather(self, irc: callbacks.NestedCommandsIrcProxy, msg: ircmsgs.IrcMsg, ar
             else:
                 irc.reply("There is an error. Contact admin.", prefixNick=False)
 
-    @wrap([optional("int"), "text"])
+    @wrap([getopts({"metric": ""}), "text"])
     def setweather(
-        self, irc: callbacks.NestedCommandsIrcProxy, msg: ircmsgs.IrcMsg, args: List[str], num: int, text: str
+        self,
+        irc: callbacks.NestedCommandsIrcProxy,
+        msg: ircmsgs.IrcMsg,
+        args: List[str],
+        optlist: List[Tuple[str, bool]],
+        text: str,
     ) -> None:
-        """<display_format> <location>
-        Sets the weather location for a user.
-        Format number is set to 1 for imperial, and 2 for metric, for which
-        units to display first. e.g. setweather 2 70118
+        """[--metric] <location>
+        Sets the weather location for a user. Format is set to show imperial first by default.
+        To show metric first, use the --metric option. e.g. setweather --metric 70118
         """
         try:
-            format_error = {"format": ["Must enter in an integer for the display format."]}
-            if not isinstance(num, int):
-                raise ValidationError(format_error)
+            format = 1
+            for (key, value) in optlist:
```

Comment:
nit: parens not necessary here.  `for key, val in optlist:` is equivalent.  

even more nitpicky: The use of the words `key` and `value` _might_ confuse a future reader when seeing "list" in the name of the iterable.  I might have used `argument` and `value`.